Update to 20w21a and DFU v3

Juuxel opened the pull request:
Mojang reworked DynamicOps (with major improvements :heart:) in DFU v3, so JanksonOps had to be half-rewritten.

I also bumped major version to 3 because it's a breaking change because of the DFU update.

Edit: also adds (de)serialization for `EntityAttribute` via the registry

Juuxel committed:
Update to 20w21a and DFU v3

Juuxel committed:
Remove broken registry serialization, add entity attribute serde

LemmaEOF commented on src/main/java/io/github/cottonmc/jankson/JanksonFactory.java:
```diff
@@ -71,24 +70,21 @@
 		register(builder, Decorator.class,          Registry.DECORATOR);
-		register(builder, DimensionType.class,      Registry.DIMENSION_TYPE);
-		
+		register(builder, EntityAttribute.class,    Registry.ATTRIBUTES);
```
don't forget to register a deserializer for this

LemmaEOF replied:
actually, wait, this class is kinda weird, just a sec

Juuxel replied:
`register` registers both a deserializer and a serializer.

LemmaEOF reviewed (changes requested):
almost there, just needs the deserializer for entity attributes

Juuxel committed:
Make all registries also have serializers

Also added all missing registries except the ones that have
a generic class like Codec or Identifier as their type.
